astropy/coordinates/representation.py

```diff
@@ -59,68 +65,112 @@ def fmt(x):
-
-        # Register representation name (except for BaseRepresentation)
-        if cls.__name__ == 'BaseRepresentation':
-            return
+def _combine_xyz(x, y, z, xyz_axis=0):
+    """
+    Combine the 3 input arrays, ``x``, ``y``, ``z`` into a single array that's
+    3 x N x ... (or N x 3 x ... or the like depending on ``xyz_axis``)
+    """
 
-        repr_name = cls.get_name()
+    # Add new axis in x, y, z so one can concatenate them around it.
+    # NOTE: just use np.stack once our minimum numpy version is 1.10.
+    result_ndim = x.ndim + 1
+    if not -result_ndim <= xyz_axis < result_ndim:
+        raise IndexError('xyz_axis {0} out of bounds [-{1}, {1})'
+                         .format(xyz_axis, result_ndim))
 
-        if repr_name in REPRESENTATION_CLASSES:
-            raise ValueError("Representation class {0} already defined".format(repr_name))
+    if xyz_axis < 0:
+        xyz_axis += result_ndim
 
-        REPRESENTATION_CLASSES[repr_name] = cls
+    # Get x, y, z to the same units (this is very fast for identical units)
+    # since np.concatenate cannot deal with quantity.
+    cls = x.__class__
+    y = cls(y, x.unit, copy=False)
+    z = cls(z, x.unit, copy=False)
 
+    sh = x.shape
+    sh = sh[:xyz_axis] + (1,) + sh[xyz_axis:]
+    xyz_value = np.concatenate([c.reshape(sh).value for c in (x, y, z)],
+                               axis=xyz_axis)
+    return cls(xyz_value, unit=x.unit, copy=False)
 
-@six.add_metaclass(MetaBaseRepresentation)
-class BaseRepresentation(ShapedLikeNDArray):
-    """
-    Base Representation object, for representing a point in a 3D coordinate
-    system.
+class RepresentationBase(ShapedLikeNDArray):
+    """3D coordinate representations and differentials.
 
-    Notes
-    -----
-    All representation classes should subclass this base representation
-    class. All subclasses should then define a ``to_cartesian`` method and a
-    ``from_cartesian`` class method. By default, transformations are done via
-    the cartesian system, but classes that want to define a smarter
-    transformation path can overload the ``represent_as`` method.
-    Furthermore, all classes must define an ``attr_classes`` attribute, an
-    `~collections.OrderedDict` which maps component names to the class that
-    creates them.  They can also define a `recommended_units` dictionary, which
-    maps component names to the units they are best presented to users in.  Note
-    that frame classes may override this with their own preferred units.
+    Parameters
+    ----------
+    comp1, comp2, comp3 : `~astropy.units.Quantity` or subclass
+        The components of the 3D point or differential.  The names are the
+        keys and the subclasses the values of the ``attr_classes`` attribute.
+    copy : bool, optional
+        If `True` (default), arrays will be copied rather than referenced.
     """
 
-    recommended_units = {}  # subclasses can override
```

Comment:
I'd actually like to remove this completely in a follow-up PR -- these recommended units are set but not used anywhere in the representations proper; this really belongs in the coordinate classes -- where presently they seem to be overridden.

But for differentials I see even less reason to have a recommended unit.